src/lib.rs

```diff
@@ -175,14 +175,29 @@ impl Decode for CENNZnutV0 {
 }
 
 impl Validate for CENNZnutV0 {
-    fn validate(&self, module_name: &str, method_name: &str) -> Result<(), &'static str> {
-        let module = self
-            .get_module(module_name)
-            .ok_or("CENNZnut does not grant permission for module")?;
-        module
-            .get_method(method_name)
-            .map(|_| ())
-            .ok_or("CENNZnut does not grant permission for method")
+    fn validate(
+        &self,
+        module_name: &str,
+        method_name: &str,
+        args: &[PactType],
+    ) -> Result<(), &'static str> {
+        self.get_module(module_name)
+            .ok_or("CENNZnut does not grant permission for module")
+            .and_then(|module| {
+                module
+                    .get_method(method_name)
+                    .ok_or("CENNZnut does not grant permission for method")
+            })
+            .and_then(|method| {
+                method
+                    .get_pact()
+                    .ok_or("CENNZnut does not grant permission for constraints")
```

Comment:
hmmm, constraints is optional so we should not error. Check `if let Some()` and try execute if so